spring-integration-core/src/main/java/org/springframework/integration/config/DefaultConfiguringBeanFactoryPostProcessor.java

```diff
@@ -172,18 +174,21 @@ private void registerMessagePublishingErrorHandler() {
 	private void registerNullChannel() {
 		if (this.beanFactory.containsBean(IntegrationContextUtils.NULL_CHANNEL_BEAN_NAME)) {
 			BeanDefinition nullChannelDefinition = null;
-			if (this.beanFactory.containsBeanDefinition(IntegrationContextUtils.NULL_CHANNEL_BEAN_NAME)) {
-				nullChannelDefinition =
-						this.beanFactory.getBeanDefinition(IntegrationContextUtils.NULL_CHANNEL_BEAN_NAME);
-			}
-			else {
-				BeanDefinitionRegistry parentBeanFactory =
-						(BeanDefinitionRegistry) this.beanFactory.getParentBeanFactory();
-				if (parentBeanFactory != null) {
-					nullChannelDefinition =
-							parentBeanFactory.getBeanDefinition(IntegrationContextUtils.NULL_CHANNEL_BEAN_NAME);
+			BeanFactory beanFactory = this.beanFactory;
+			do {
+				if (beanFactory instanceof ConfigurableListableBeanFactory) {
+					ConfigurableListableBeanFactory listable = (ConfigurableListableBeanFactory) beanFactory;
+					if (listable.containsBeanDefinition(IntegrationContextUtils.NULL_CHANNEL_BEAN_NAME)) {
+						nullChannelDefinition = listable
+								.getBeanDefinition(IntegrationContextUtils.NULL_CHANNEL_BEAN_NAME);
+					}
 				}
-			}
+				// don't think it will hurt to always find the parent
+				if (beanFactory instanceof HierarchicalBeanFactory) {
+					beanFactory = ((HierarchicalBeanFactory) beanFactory).getParentBeanFactory();
+				}
+				// will definitely be found as containsBean returned true - but also want to be defensive in case of NPE
+			} while (nullChannelDefinition == null && beanFactory != null); // not sure if beanFactroy not null is necessary
```

Comment:
The `while` must be on a new line. 
I think we indeed don't need to check for `beanFactory != null`, since you are right that `containsBean(IntegrationContextUtils.NULL_CHANNEL_BEAN_NAME)` does the trick for us do not fail with the infinite loop. 
Although if Checkstyle or Sonar Qube will require it I would just live with an extra check in this condition.

Reply:
Yes, I'll ensure that the coding style is adhered to. Finding meeting coding guidelines quite challenging

Reply:
comments and null check on bean factory removed.